experimental/segmenter/build.rs

```diff
@@ -0,0 +1,404 @@
+            // Use common table
+            if is_same_value && *c == 0 {
+                codepoint_table.pop();
+                codepoint_table.push("BREAK_PROPERTIES_FILL_BY_0".to_string());
+            }
+
+            codepoint_table.push(format!("BREAK_PROPERTIES_{}", page));
+            writeln!(out, "#[allow(dead_code)]").ok();
+            writeln!(out, "pub const BREAK_PROPERTIES_{}: [u8; 1024] = [", page).ok();
+            continue;
+        }
+    }
+
+    writeln!(out, "#[allow(dead_code)]").ok();
+    writeln!(out, "pub const BREAK_PROPERTIES_FILL_BY_0: [u8; 1024] = [").ok();
+    for i in 0..1024 {
+        write!(out, " 0,").ok();
+        if ((i + 1) % 16) == 0 {
+            writeln!(out).ok();
+        }
+    }
+    writeln!(out, "];").ok();
+
+    writeln!(out, "pub const PROPERTY_TABLE: [&[u8; 1024]; 192] = [").ok();
+    for i in codepoint_table.iter() {
+        writeln!(out, "    &{},", i).ok();
+    }
+    writeln!(out, "];").ok();
+
+    writeln!(
+        out,
+        "pub const BREAK_STATE_MACHINE_TABLE: [i8; {}] = [",
+        rule_size
+    )
+    .ok();
+    let mut i = 1;
+    writeln!(out, "// {}", properties_names[i - 1]).ok();
+    for c in break_state_table.iter() {
+        write!(out, "{: >4},", c).ok();
+        i += 1;
+        if ((i - 1) % properties_names.len()) == 0 {
+            writeln!(out).ok();
+            if (i / properties_names.len()) < properties_names.len() {
+                writeln!(out, "// {}", properties_names[i / properties_names.len()]).ok();
+            }
+        }
+    }
+    writeln!(out, "];").ok();
+
+    writeln!(
+        out,
+        "pub const PROPERTY_COUNT: usize = {};",
+        properties_names.len()
+    )
+    .ok();
+    writeln!(
+        out,
+        "pub const LAST_CODEPOINT_PROPERTY: i8 = {};",
+        simple_properties_count - 1
+    )
+    .ok();
+
+    writeln!(
+        out,
+        "pub const PROP_SOT: usize = {};",
+        properties_names.len() - 2
+    )
+    .ok();
+    writeln!(
+        out,
+        "pub const PROP_EOT: usize = {};",
+        properties_names.len() - 1
+    )
+    .ok();
+    if let Some(sa_index) = get_index_from_name(&properties_names, "SA") {
+        writeln!(out, "pub const PROP_COMPLEX: usize = {};", sa_index,).ok();
+    }
+
+    for (i, p) in properties_names.iter().enumerate() {
+        writeln!(out, "// {} = {}", p, i).ok();
+    }
+
+    writeln!(out).ok();
+    writeln!(out, "#[allow(dead_code)]").ok();
+    writeln!(out, "pub const BREAK_RULE: i8 = -128;").ok();
+    writeln!(out, "#[allow(dead_code)]").ok();
+    writeln!(out, "pub const NOT_MATCH_RULE: i8 = -2;").ok();
+    writeln!(out, "#[allow(dead_code)]").ok();
+    writeln!(out, "pub const KEEP_RULE: i8 = -1;").ok();
+}
+
+fn main() {
+    const WORD_SEGMENTER_JSON: &[u8] = include_bytes!("data/word.json");
+
+    let provider = icu_testdata::get_provider();
+    generate_rule_segmenter_table("generated_word_table.rs", WORD_SEGMENTER_JSON, &provider);
```

Comment:
Question: How big is the data inside "generated_word_table.rs", as in, how much static memory space does it consume (and, likewise, what is its impact on binary size)?

Reply:
Actually, 256KB+8KB+ 2KB in binary at least, generated rs file is 3MB.